File: android/app/src/main/java/com/moviestars/modules/logger/LoggerModule.kt
package com.moviestars.modules.logger

import android.util.Log
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import javax.inject.Inject

class LoggerModule(reactContext: ReactApplicationContext?) :
    ReactContextBaseJavaModule(reactContext)  {

    var TAG = "LoggerModule"

    @Inject lateinit var sentryLogger: SentryLogger

    override fun getName(): String = "LoggerModule"

    @ReactMethod
    fun log(message: String) {
        Log.d(TAG, message);
        sentryLogger.log(message)
    }

    @ReactMethod
    fun error(message: String, error: Throwable) {
        Log.d(TAG, message, error);
        sentryLogger.error(message, error)
    }

}

File: ios/modules/logger/LoggerModule.swift
//
//  LoggerModule.swift
//  moviestars
//
//  Created by Adriano Jesus on 19/06/23.
//

import Foundation

@objc(LoggerModule)
class LoggerModule: NSObject {
  
  let sentryLogger = SentryLogger();
  
  @objc(log:)
  func log(_ message: String) {
    self.sentryLogger.log(message: message)
  }
  
  @objc(error:error:)
  func error(_ message: String, error: Error) {
    self.sentryLogger.error(message: message, error: error)
  }

}
